experimental/devices/cap1188/cap1188.go

```diff
@@ -156,369 +121,396 @@ func (d *Dev) InputStatus() ([]TouchStatus, error) {
-	if opts != nil {
-		switch opts.Address {
-		case 0x28, 0x29, 0x2a, 0x2b, 0x2c:
-			addr = opts.Address
-		case 0x00:
-			// do not do anything
-		default:
-			return nil, wrap(errors.New("given address not supported by device"))
-		}
+	if opts == nil {
+		opts = DefaultOpts()
 	}
-	d := &Dev{d: &i2c.Dev{Bus: b, Addr: addr}, isSPI: false}
-	if d.Debug {
-		log.Printf("cap1188: Connecting via I2C address: %#X\n", addr)
+	addr, err := opts.i2cAddr()
+	if err != nil {
+		return nil, wrapf("%v", err)
 	}
-	d.inputStatuses = make([]TouchStatus, nbrOfLEDs)
-	if err := d.makeDev(opts); err != nil {
+	d, err := makeDev(&i2c.Dev{Bus: b, Addr: addr}, false, opts)
+	if err != nil {
 		return nil, err
 	}
-	// time to communications is 15ms
-	now := time.Now()
-	readyAt := d.resetAt.Add(15 * time.Millisecond)
-	if now.Before(readyAt) {
-		time.Sleep(readyAt.Sub(now))
-	}
+	log.Printf("cap1188: Connected via I²C address: %#x", addr)
 	return d, nil
 }
 
+/*
 // NewSPI returns an object that communicates over SPI to cap1188 environmental
 // sensor.
+//
+// TODO(mattetti): Expose once implemented and tested.
 func NewSPI(p spi.Port, opts *Opts) (*Dev, error) {
-	return nil, fmt.Errorf("not implemented")
+	return nil, fmt.Errorf("cap1188: not implemented")
 }
+*/
 
-func (d *Dev) makeDev(opts *Opts) error {
-	// Use default options if none are passed.
-	if opts == nil {
-		opts = DefaultOpts()
+//
+
+func makeDev(c conn.Conn, isSPI bool, opts *Opts) (*Dev, error) {
+	d := &Dev{
+		opts:  *opts,
+		isSPI: isSPI,
+		c:     mmr.Dev8{Conn: c, Order: binary.LittleEndian},
 	}
-	d.Opts = *opts
-	d.regWrapper = mmr.Dev8{Conn: d.d, Order: binary.LittleEndian}
 
-	var productID byte
-	var err error
 	// Read the product id to confirm it matches our expectations.
-	if productID, err = d.regWrapper.ReadUint8(0xFD); err != nil {
-		return fmt.Errorf("failed to read product id - %s", err)
-	}
-	if productID != 0x50 {
-		return fmt.Errorf("cap1188: unexpected chip id %x; is this a cap1188?", productID)
+	if productID, err := d.c.ReadUint8(0xFD); err != nil {
+		return nil, wrapf("failed to read product id: %v", err)
+	} else if productID != 0x50 {
+		return nil, wrapf("unexpected chip id %x; is this a cap1188?", productID)
 	}
 	// manufacturer ID on 0xFE, should be 0x5D
 	// revision ID on 0xFF, should be 0x83
 
-	// reset the device
-	if err = d.Reset(); err != nil {
-		return fmt.Errorf("failed to reset the device - %s", err)
+	// Reset the device.
+	if err := d.Reset(); err != nil {
+		return nil, err
 	}
 
 	var recalFlag byte
-	if opts.EnableRecalibration {
+	if d.opts.EnableRecalibration {
 		recalFlag = 1
 	}
 	var intOnRel byte
-	if !opts.InterruptOnRelease {
+	if !d.opts.InterruptOnRelease {
 		intOnRel = 1 // 0 = trigger on release
 	}
 
-	// enable all inputs
-	if err = d.regWrapper.WriteUint8(0x21, 0xff); err != nil {
-		return wrap(fmt.Errorf("failed to enable all inputs - %s", err))
+	// Enable all inputs.
+	if err := d.c.WriteUint8(0x21, 0xff); err != nil {
```

Comment:
is there an advantage of allocating a new local variable err for each of these conditions vs declaring and reusing the same one?

Reply:
It's now needed because there is no more a function local err declared above.